server/src/main/java/io/deephaven/server/arrow/ArrowFlightUtil.java

```diff
@@ -394,24 +393,31 @@ interface Handler extends Closeable {
         public DoExchangeMarshaller(
                 final TicketRouter ticketRouter,
                 final BarrageMessageWriter.Factory streamGeneratorFactory,
-                final BarrageMessageProducer.Operation.Factory bmpOperationFactory,
-                final HierarchicalTableViewSubscription.Factory htvsFactory,
-                final BarrageMessageProducer.Adapter<BarrageSubscriptionRequest, BarrageSubscriptionOptions> subscriptionOptAdapter,
-                final BarrageMessageProducer.Adapter<BarrageSnapshotRequest, BarrageSnapshotOptions> snapshotOptAdapter,
+                final Set<ExchangeMarshaller> exchangeMarshallers,
+                final Set<ExchangeRequestHandlerFactory> requestHandlerFactories,
                 final SessionService.ErrorTransformer errorTransformer,
                 @Assisted final SessionState session,
                 @Assisted final StreamObserver<InputStream> responseObserver) {
 
             this.myPrefix = "DoExchangeMarshaller{" + Integer.toHexString(System.identityHashCode(this)) + "}: ";
             this.ticketRouter = ticketRouter;
             this.streamGeneratorFactory = streamGeneratorFactory;
-            this.bmpOperationFactory = bmpOperationFactory;
-            this.htvsFactory = htvsFactory;
-            this.subscriptionOptAdapter = subscriptionOptAdapter;
-            this.snapshotOptAdapter = snapshotOptAdapter;
             this.session = session;
             this.listener = new MessageViewAdapter(responseObserver);
             this.errorTransformer = errorTransformer;
+            this.marshallers = new ArrayList<>(exchangeMarshallers.size());
+            marshallers.addAll(exchangeMarshallers);
```

Comment:
Maybe instead just
```suggestion
            this.marshallers = new ArrayList<>(exchangeMarshallers);
```
or stream and sort... not quite a one liner though.

Could also make a dagger `@Provides` method that takes a set and returns a singleton list so don't copy and sort every time a subscription is created.